server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -293,63 +287,44 @@ public void intersectLeft(PointValues.IntersectVisitor visitor, PointValues.Poin
                 }
             }
 
-            // custom intersect visitor to walk the right of tree
+            // custom intersect visitor to walk the right of tree (from rightmost leaf going left)
             public void intersectRight(PointValues.IntersectVisitor visitor, PointValues.PointTree pointTree, long[] docCount)
                 throws IOException {
-                PointValues.Relation r = visitor.compare(pointTree.getMinPackedValue(), pointTree.getMaxPackedValue());
                 if (docCount[0] >= size) {
                     return;
                 }
+                PointValues.Relation r = visitor.compare(pointTree.getMinPackedValue(), pointTree.getMaxPackedValue());
                 switch (r) {
-                    case CELL_OUTSIDE_QUERY:
-                        // This cell is fully outside the query shape: stop recursing
-                        break;
-
                     case CELL_INSIDE_QUERY:
-                        // If the cell is fully inside, we keep moving right as long as the point tree size is over our size requirement
-                        if (pointTree.size() > size && docCount[0] < size && moveRight(pointTree)) {
+                    case CELL_CROSSES_QUERY:
+                        if (pointTree.moveToChild() && docCount[0] < size) {
```

Comment:
Remove `docCount[0] < size`, already checked at the beginning of the method

Reply:
I dont see it a problem, this check prevents processing children when we've collected enough documents during the current method execution.

Reply:
Do you mean this check is needed here? I think Line 293-295 already check this, or I'm missing sth?